guidelines/techpreview.md

```diff
@@ -0,0 +1,97 @@
+
+
+# Downsides to declaring something Tech Preview
+* Since your feature requires special action to enable, you won’t get default CI coverage.  You may need to introduce your own 
+CI job that enables the TP feature if you want automated coverage
+* To date we have seen very few customers enabling feature gates (in part because they block upgrading that cluster) so if your
+feature is behind the cluster feature gate, you are unlikely to get meaningful feedback from the field to help you evolve the 
+feature anyway.  It may be better to just hold the feature until it’s GA ready.
+
+
+# Official process/mechanism for delivering a TP feature
+
+1. Your feature must be disabled by default
+1. Enabling it must require the admin enable a specific feature gate such as [TechPreviewNoUpgrade](https://docs.openshift.com/container-platform/4.1/nodes/clusters/nodes-cluster-enabling-features.html)
+1. You need to list the feature in [this set](https://github.com/openshift/api/blob/bace76a807222b30bb9bfd4926826348156fb522/config/v1/types_feature.go#L117)
+1. Your operator then needs to observe whether the feature gate is enabled and then, and only then, can it enable the feature 
+meaning installing any TP CRDs and performing the TP behavior.
+1. optional:  you can have additional mechanisms for enabling the TP feature that are checked in addition to the cluster-scoped 
+feature gate mechanism, but you must have the feature gate mechanism.
+1. optional:  if your feature gate is not enabled and the TP fields are populated by the user it is recommended that your
+component should clear that data from the fields to avoid user confusion when they think they’ve configured the feature but 
+it’s not actually active/enabled.
+
+
+## Following this process means
+
+* No customer will be able use your feature by default
+* If they do enable it, they cannot upgrade their cluster, so very few will use it.  You will not likely get much user feedback 
+on the feature if that is your goal.
+* Your feature will not be available in CI clusters unless you create your own specific CI job that enables the featuregate so 
+you can test the feature.
+* You must not install any CRDs related to the TP feature unless the TP is enabled.
+
+
+
+# Unofficial Processes/Mechanisms
+
+## Option 1
+
+If you are confident in the api (or your TP Feature does not require api changes), and you are prepared to allow upgrades for 
+customers who enabled this feature (you do not have to keep the TP feature working during/after the upgrade, but you have to 
+allow the upgrade and not fail/crashloop/etc) you can take a more minimal approach to exposing your feature as TP by
+
+1. Requiring the admin to set some field that contains TechPreview to true, such as “EnableTechPreviewFooBar=true” to enable your 
+feature
```

Comment:
If an entire OLM operator is TechPreview, is documenting it as TechPreview sufficient?  The admin already must take a deliberate action to install it, so setting a field like this somewhere doesn't really seem necessary.

Reply:
I don't know how common that is, but yes that seems reasonable to me, as long as we are confident that the admin is confronted w/ the fact that the operator is Tech Preview as part of the operator selection/install process.

Reply:
adding this bullt:
```
For OLM operators where the entire operator is considered Tech Preview, clearly describing the operator as Tech Preview(or using a channel with TechPreview in the name) is sufficient as the admin must still take an explicit action to install/enable your operator.
```

Reply:
I also wouldn't expect the installation of a tech preview optional operator to necessarily require the whole cluster be un-upgradeable. Maybe in some exceptional cases, but for the most part installing these operators shouldn't impact the underlying clusters ability to upgrade itself.

Reply:
@jwforres so this option is listed under the section described as:
```
If you are confident in the API (or your TP Feature does not require API changes), and you are prepared to allow upgrades for 
customers who enabled this feature (you do not have to keep the TP feature working during/after the upgrade, but you have to 
allow the upgrade and not fail/crashloop/etc) you can take a more minimal approach to exposing your feature as TP
```

so are you just agreeing w/ the new bullet(1.1), or were you looking for a different statement?